server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -145,28 +146,70 @@ public class BalancedShardsAllocator implements ShardsAllocator {
         Property.NodeScope
     );
 
+    public static final Setting<Boolean> PREFER_PRIMARY_SHARD_REBALANCE = Setting.boolSetting(
+        "cluster.routing.allocation.rebalance.prefer_primary",
+        false,
+        Property.Dynamic,
+        Property.NodeScope
+    );
+
+    public static final Setting<Float> PRIMARY_SHARD_REBALANCE_BUFFER = Setting.floatSetting(
+        "cluster.routing.allocation.rebalance.primary_buffer",
+        0.05f,
```

Comment:
Should we have a higher default value for low to no impact in relocation?

Reply:
Ack.